editor/src/components/canvas/commands/push-intended-bounds-and-update-groups-command.ts

```diff
@@ -134,24 +137,38 @@ function getUpdateResizedGroupChildrenCommands(
       frameAndTarget.target,
     )
     if (targetIsGroup) {
-      const originalSize: Size = sizeFromRectangle(
-        MetadataUtils.getLocalFrameFromSpecialSizeMeasurements(
-          frameAndTarget.target,
-          editor.jsxMetadata,
-        ),
+      const children = MetadataUtils.getChildrenPathsOrdered(
+        editor.jsxMetadata,
+        editor.elementPathTree,
+        frameAndTarget.target,
       )
+
+      const originalSize: Size =
+        command.isMetadataStale === 'metadata-is-stale'
+          ? sizeFromRectangle(
+              MetadataUtils.getLocalFrameFromSpecialSizeMeasurements(
+                frameAndTarget.target,
+                editor.jsxMetadata,
+              ),
+            )
+          : sizeFromRectangle(
+              boundingRectangleArray(
+                children.map((c) =>
+                  nullIfInfinity(
+                    MetadataUtils.findElementByElementPath(editor.jsxMetadata, c)?.globalFrame,
+                  ),
+                ),
+              ),
+            )
```

Comment:
I really don't understand why if the metadata is stale we're using the special size measurements of the target, but if the metadata is fresh we're using the bounding rectangle of the children. Please can you add a comment here clarifying why

Reply:
I've pushed comments. if you have any suggestions that are better than 'metadata-is-stale' I'm happy to go for it